lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,472 @@
+	}
+}
+
+// getKubeconfig returns kubeconfig that can be used to access target EKS cluster
+func getKubeconfig(ctx context.Context, timestamp time.Time, credsProvider aws.CredentialsProvider, clusterName, region, clusterCA, clusterEndpoint string) (*genericclioptions.ConfigFlags, error) {
+	targetUrl := getPresignURL()
+
+	r, err := http.NewRequest(http.MethodGet, targetUrl.String(), nil)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	r.Header.Add(awsHeaderClusterName, clusterName)
+	creds, err := credsProvider.Retrieve(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	signer := v4.NewSigner()
+	presigned, _, err := signer.PresignHTTP(ctx, creds, r, hashForGetRequests, "sts", region, timestamp)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	kubeToken := awsKubePrefix + base64.RawURLEncoding.EncodeToString([]byte(presigned))
+
+	eksClusterCA, err := base64.StdEncoding.DecodeString(clusterCA)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	restConfig := &rest.Config{
+		Host:        clusterEndpoint,
+		BearerToken: kubeToken,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: eksClusterCA,
+		},
+	}
+
+	kubeConfig := genericclioptions.NewConfigFlags(false)
+	kubeConfig.APIServer = &restConfig.Host
+	kubeConfig.BearerToken = &restConfig.BearerToken
+	kubeConfig.WithWrapConfigFn(func(*rest.Config) *rest.Config {
+		return restConfig
+	})
+
+	return kubeConfig, nil
+}
+
+func getHelmActionConfig(kubeConfig genericclioptions.RESTClientGetter, log logrus.FieldLogger) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(kubeConfig, agentNamespace, "secret", log.WithField("source", "helm").Debugf); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return actionConfig, nil
+}
+
+func getHelmSettings() *helmCli.EnvSettings {
+	helmSettings := helmCli.New()
+	helmSettings.RepositoryCache = os.TempDir()
+	helmSettings.SetNamespace(agentNamespace)
+
+	return helmSettings
+}
+
+// checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
+func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
+	listCmd := action.NewList(actionConfig)
+
+	releases, err := listCmd.Run()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	for _, release := range releases {
+		if release.Name == agentName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// installKubeAgent installs teleport-kube-agent chart to the target EKS cluster.
+func installKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr string, actionConfig *action.Configuration, settings *helmCli.EnvSettings, req EnrollEKSClustersRequest) error {
+	installCmd := action.NewInstall(actionConfig)
+	installCmd.RepoURL = agentRepoURL
+	installCmd.Version = teleport.Version
+	if strings.Contains(teleport.Version, "dev") {
+		installCmd.Version = "14.3.0" // For testing during development.
+	}
```

Comment:
We should get the current version from the configured auto-upgrade channel.
That will probably give you 14.3.X if you have a ent+cloud license

My point is, when cluster is enrolled into auto-upgrades, not only should we enable auto-upgrades in the agent but we should also install the current version (and not `teleport.version`)

Reply:
I've refactored it to use configure automatic upgrades channel to get the version - https://github.com/gravitational/teleport/pull/36594/commits/9a089d01f9d486eaf2181d91ab1576c5f0df8433